s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -136,9 +136,10 @@ func (t *TidbClient) GetBucket(ctx context.Context, bucketName string) (bucket *
 }
 
 // For the request that list buckets, better to filter according to tenant.
-func (t *TidbClient) GetBuckets(ctx context.Context) (buckets []*Bucket, err error) {
-	log.Info("list buckets from tidb ...")
-
+func (t *TidbClient) GetBuckets(ctx context.Context, query interface{}) (buckets []*Bucket, err error) {
+	log.Info("list buckets from tidb ...\n")
+	filter := query.(map[string]string)
+	filterLoc := filter["location"]
```

Comment:
why hardcoded "location"?
what happens, if some other filter options come later.

Reply:
will be made into subfunction